flask/datahandle.py

```diff
@@ -5,6 +5,32 @@
 
 ###Helper functions
 
+
+def formatItem(item):
+    """Removes excess ending spaces/commas.
+
+    :param item: The string to be manipulated
+    :type item: string
+    """
+    while(item[-1] == ' ' or item[-1] == ','):
+      item = item[:-1]
+    return item
+
+def formatPlural(text):
+    """Adds 'and' before last item in list of items.
+
+    :param text: The string to be manipulated
+    :type text: string
+    """
+    if ',' in text:
+        for i in range(len(text)):
```

Comment:
Rather than doing this loop you I believe rfind can accomplish this. Can also check if index != -1. 
```
index = text.rfind(',') + 1
#This will find the location of the last comma). Then just insert your add in there. 
text = text[:index] + 'and ' + text[index:]
```